infrastructure/logging-bucket.yaml

```diff
@@ -0,0 +1,32 @@
+# Define a Log Bucket in GCP to store application logs
+apiVersion: logging.cnrm.cloud.google.com/v1beta1
+kind: LoggingLogBucket
+metadata:
+  name: hopic-application-logs-pht-01hp04dtnkf
+  namespace: cnrm-system
+spec:
+  projectRef:
+    external: "projects/pht-01hp04dtnkf"
+  location: "northamerica-northeast1"
+  description: "Hopic Application Logs Storage"
+  # Whether the log bucket is protected from deletion
+  locked: true
+  # Number of days to retain logs in the bucket
+  retentionDays: 90
+---
+# Define a Log Sink for routing logs to the defined Log Bucket
+apiVersion: logging.cnrm.cloud.google.com/v1beta1
+kind: LoggingLogSink
+metadata:
+  namespace: cnrm-system
+  name: hopic-application-logs-sink
+spec:
+  projectRef:
+    external: "pht-01hp04dtnkf"
+  # Destination bucket for the logs captured by this sink
+  destination:
+    loggingLogBucketRef:
+      external: "logging.googleapis.com/projects/pht-01hp04dtnkf/locations/northamerica-northeast1/buckets/hopic-application-logs-pht-01hp04dtnkf"
```

Comment:
Since the `LoggingLogBucket` resource is a part of KCC, it'd be helpful to use `name` and `namespace` keys to refer to the resource instead of an external link.

Reply:
@vedantthapa Similar issue here as well, I tried with using the bucket name and I receive the error.

----     ------        ----  ----                       -------                                                                                                                      │
│   Warning  UpdateFailed  1s    logginglogsink-controller  Update call failed: error fetching live state: error converting resource config: error getting referenced resource from API  │
│ server: error getting ResourceConfig for referenced resource hopic-application-logs-sink: couldn't find any ResourceConfig defined for kind LoggingLogBucket      

Here is the issue described.

https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/665

Reply:
This is a known limitation which is [documented](https://cloud.google.com/config-connector/docs/reference/resource-docs/logging/logginglogsink#:~:text=Only%20%60external%60%20field%20is%20supported%20to%20configure%20the%20reference.) in our current resource reference.

Issue: https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/665